index.js

```diff
@@ -127,8 +128,17 @@ class Hydra extends EventEmitter {
           this.serviceDescription = this.config.serviceDescription || 'not specified';
           this.serviceVersion = this.config.serviceVersion || 'not specified';
 
+          // if serviceIP field contains a name rather than a dotted IP address
+          // then use DNS to resolve the name to an IP address.
+          if (this.config.serviceIP &&
+              this.config.serviceIP !== '' &&
+              this.config.serviceIP.indexOf('.') === -1) {
```

Comment:
I think this check should be more robust, since hosts can contain dots too.

Maybe something like this:
```javascript
let isNumericIP = (serviceIP) => {
  let parts = serviceIP.split('.')
  if (parts.length !== 4) {
    return false;
  }
  return parts.filter(part => {
    if (Number.isNaN(part)) {
      return true;
    }
    let num = Number(part);
    if (num < 0 || num > 255) {
      return true;
    }
  }).length ? false : true;
};
```